src/v2/cookbook/packaing-sfc-for-npm.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Packaging Vue Components for npm
+type: cookbook
+order: 10
+---
+
+## Base Example
+
+Vue components by nature are meant to be re-used. This is easy when the component is only used within a single application. But how can you write a component once and use it in multiple sites/applications? Perhaps the easiest solution is via npm.
+
+By packaging your component to be shared via npm, it can be imported/required into a build process for use in full-fledged web applications:
+
+```js
+import MyComponent from 'my-component';
+
+export default {
+  components: {
+    MyComponent,
+  },
+  // rest of the component
+}
+```
+
+Or even used via `<script>` tag in the browser directly:
+
+```html
+  <script src="https://unpkg.com/vue"></script>
+  <script src="https://unpkg.com/my-component"></script>
+  ...
+  <my-component></my-component>
+  ...
+```
+
+Not only does this help you avoid copy/pasting components around, but it also allows you to give back to the Vue community!
+
+## Can't I Just Share `.vue` Files Directly?
+
+Vue already allows components to be written as a single file. Because a Single File Component (SFC) is already just one file, you might ask:
+
+> "Why can't people use my `.vue` file directly? Isn't that the simplest way to share components?"
+
+It's true, you can share `.vue` files directly, and anyone using a [Vue build](https://vuejs.org/v2/guide/installation.html#Explanation-of-Different-Builds) containing the Vue compiler can consume it immediately. Also, the SSR build uses string concatenation as an optimization, so the `.vue` file might be preferred in this scenario (see [Packaging Components for npm > SSR Usage](#SSR-Usage) for details). However, this excludes anyone who wishes to use the component directly in a browser via `<script>` tag, anyone who uses a runtime-only build, or build processes which don't understand what to do with `.vue` files.
+
+Properly packaging your SFC for distribution via npm enables your component to be shared in a way which is ready to use everywhere!
+
+## Packaging Components for npm
+
+For the purposes of this section, assume the following file structure:
```

Comment:
this type of explanation of structure before diving in is really useful to people reading 👍